File: Frontend/flutter_app/lib/class/car.dart
import 'car_price.dart';

class Car {
  final int carId;
  String brand;
  String model;
  String category;
  int year;
  String status;
  String? imageUrl;             
  List<CarPrice> prices;       

  Car({
    required this.carId,
    required this.brand,
    required this.model,
    required this.category,
    required this.year,
    required this.status,
    this.imageUrl,
    List<CarPrice>? prices,
  }) : prices = prices ?? [];


  String getBrand() => brand;
  void setBrand(String v) => brand = v;

  String getModel() => model;
  void setModel(String v) => model = v;

  String getCategory() => category;
  void setCategory(String v) => category = v;

  int getYear() => year;
  void setYear(int v) => year = v;

  String getStatus() => status;
  void setStatus(String v) => status = v;

  String? getImageUrl() => imageUrl;
  void setImageUrl(String? v) => imageUrl = v;

  void addPrice(CarPrice p) => prices.add(p);

  double? getPriceForDay(String day) {
    for (final p in prices) {
      if (p.day == day) return p.price;
    }
    return null;
  }

  List<double> getPricesOnly() => prices.map((p) => p.price).toList();


factory Car.fromJson(Map<String, dynamic> json) {
  return Car(
    carId: json['car_id'],
    brand: json['brand'],
    model: json['model'],
    category: json['category'],
    year: json['year'],
    status: json['status'],
    imageUrl: json['image_url'],
    prices: json['prices']
        .map((p) => CarPrice.fromJson(p))
        .toList(),
  );
}

  Map<String, dynamic> toJson() {
    return {
      'car_id': carId,
      'brand': brand,
      'model': model,
      'category': category,
      'year': year,
      'status': status,
      'image_url': imageUrl,
      'prices': prices.map((p) => p.toJson()).toList(),
    };
  }
}

File: Backend/schemas/car.py

from pydantic import BaseModel
from typing import Optional, List
from schemas.car_price import CarPriceOut, CarPriceCreate

class CarOut(BaseModel):
    car_id: int
    brand: str
    model: str
    category: str
    year: int
    status: bool
    image_url: Optional[str] = None
    prices: List[CarPriceOut] = []


class CarCreate(BaseModel):
    brand: str
    model: str
    category: str
    year: int
    status: bool = True
    image_url: Optional[str] = None
    prices: List[CarPriceCreate] = []

class CarUpdate(BaseModel):
    brand: Optional[str] = None
    model: Optional[str] = None
    category: Optional[str] = None
    year: Optional[int] = None
    status: Optional[bool] = None   
    image_url: Optional[str] = None
    prices: List[CarPriceCreate] = []